src/Lean/Server/InfoUtils.lean

```diff
@@ -299,36 +301,9 @@ where
       cs.any (hasNestedTactic pos tailPos)
     | _ => false
 
-/--
-Find info nodes that should be used for the term goal feature.
-
-The main complication concerns applications
-like `f a b` where `f` is an identifier.
-In this case, the term goal at `f`
-should be the goal for the full application `f a b`.
-
-Therefore we first gather the position of
-these head function symbols such as `f`,
-and later ignore identifiers at these positions.
--/
 partial def InfoTree.termGoalAt? (t : InfoTree) (hoverPos : String.Pos) : Option (ContextInfo × Info) :=
-  let headFns : Std.HashSet String.Pos := t.foldInfo (init := {}) fun _ i headFns =>
-    if let some pos := getHeadFnPos? i.stx then
-      headFns.insert pos
-    else
-      headFns
-  t.smallestInfo? fun i => Id.run do
-    if i.contains hoverPos then
-      if let Info.ofTermInfo ti := i then
-        return !ti.stx.isIdent || !headFns.contains i.pos?.get!
-    false
-  where
-    /- Returns the position of the head function symbol, if it is an identifier. -/
-    getHeadFnPos? (s : Syntax) (foundArgs := false) : Option String.Pos :=
-      match s with
-      | `(($s)) => getHeadFnPos? s foundArgs
```

Comment:
I assume this was to make `(f a) b` behave like `f a b`, but is that really intuitive? I could even see myself temporarily replacing `f` with `(f)` just to see the instantiated by unapplied type.

Reply:
Your assumption is correct.  It feels a bit counterintuitive that inserting parentheses changes the output (when they're usually no-ops only used to disambiguate precedence).  I'm fine with either version though.

Reply:
Let's wait and see then